### PR TITLE
Reduce slightly scanpy tools profile for wider instance support

### PR DESCRIPTION
**Pull request by pcm32**
FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] License permits unrestricted use (educational + commercial)
* [ ] This PR adds a new tool or tool collection
* [ ] This PR updates an existing tool or tool collection
* [x] This PR does something else (explain below)

The IUC Scanpy tools are using a profile version that doesn't allow us to install them alongside our tools on some private instances that we have. I noticed that the current profile had a value outside of the described ones on the [tools XML docs](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-profile), so I have nudged the value to the closest lower one.

This doesn't have any impact on tools really, mostly on where they can be installed, but let me know if this merits a `galaxy+<num>` version bump on tools. I hope that this change is acceptable.

### CODE REVIEW

**Comment by pcm32** [comment]
Would it be possible to waive the old/broken URLs updates? This makes updating any old tool (and not so old by the looks of it) very laborious :-(.

**Comment by pavanvidem** [comment]
how about changing all the links to https://scanpy.readthedocs.io? For eg. https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pp.neighbors.html works

**Comment by pcm32** [comment]
That didn't cut it :-(...

**Comment by pcm32** [comment]
So it is a matter of chasing link by link, which is really a lot of time (I tried a few easy iterations on the current ones without success).... I would just remove them, people if needed can always search for scanpy's documentation and find the current one (actually the tool points to a different version of the documentation that it should, so it is not completely correct from purist point of view, but chasing that is a lot of energy). Or suggest the change in the linter to waive broken URLs if possible.

File: tools/scanpy/plot.xml [hunk 1]
```diff
@@ -2329,33 +2329,33 @@ plotted as boxplots.
 This plot is similar to the `scater` package function `plotHighestExprs(type= "highest-expression")`
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.highest_expr_genes.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.highest_expr_genes.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2332]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.highest_expr_genes.html>`__
```

File: tools/scanpy/plot.xml [hunk 2]
```diff
@@ -2329,33 +2329,33 @@ plotted as boxplots.
...
 
 Preprocessing: Plot dispersions versus means for genes (`pl.highly_variable_genes`)
 ===================================================================================
 
 It produces Supp. Fig. 5c of Zheng et al. (2017) and MeanVarPlot() of Seurat.
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.highly_variable_genes.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.highly_variable_genes.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2340]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.highly_variable_genes.html>`__
```

File: tools/scanpy/plot.xml [hunk 3]
```diff
@@ -2329,33 +2329,33 @@ plotted as boxplots.
...
 
 PCA: Scatter plot in PCA coordinates (`pl.pca`)
 ===============================================
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.pca.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.pca.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2346]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.pca.html>`__
```

File: tools/scanpy/plot.xml [hunk 4]
```diff
@@ -2329,33 +2329,33 @@ plotted as boxplots.
...
 
 PCA: Rank genes according to contributions to PCs (`pl.pca_loadings`)
 =====================================================================
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.pca_loadings.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.pca_loadings.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2352]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.pca_loadings.html>`__
```

File: tools/scanpy/plot.xml [hunk 5]
```diff
@@ -2329,33 +2329,33 @@ plotted as boxplots.
...
 
 PCA: Plot the variance ratio (`pl.pca_variance_ratio`)
 ======================================================
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.pca_variance_ratio.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.pca_variance_ratio.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2358]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.pca_variance_ratio.html>`__
```

File: tools/scanpy/plot.xml [hunk 6]
```diff
@@ -2364,37 +2364,37 @@ The parameters are the ones of the scatter plot. Call pca_ranking separately
 if you want to change the default settings.
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.pca_overview.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.pca_overview.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2367]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.pca_overview.html>`__
```

File: tools/scanpy/plot.xml [hunk 7]
```diff
@@ -2364,37 +2364,37 @@ The parameters are the ones of the scatter plot. Call pca_ranking separately
...
 
 Embedding: Scatter plot in tSNE basis (`pl.tsne`)
 =================================================
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.tsne.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.tsne.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2373]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.tsne.html>`__
```

File: tools/scanpy/plot.xml [hunk 8]
```diff
@@ -2364,37 +2364,37 @@ The parameters are the ones of the scatter plot. Call pca_ranking separately
...
 
 Embeddings: Scatter plot in UMAP basis (`pl.umap`)
 ==================================================
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.umap.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.umap.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2379]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.umap.html>`__
```

File: tools/scanpy/plot.xml [hunk 9]
```diff
@@ -2364,37 +2364,37 @@ The parameters are the ones of the scatter plot. Call pca_ranking separately
...
 
 Embeddings: Scatter plot in Diffusion Map basis (`pl.diffmap`)
 ==============================================================
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.diffmap.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.diffmap.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2385]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.diffmap.html>`__
```

File: tools/scanpy/plot.xml [hunk 10]
```diff
@@ -2364,37 +2364,37 @@ The parameters are the ones of the scatter plot. Call pca_ranking separately
...
 
 Branching trajectories and pseudotime, clustering: Plot groups and pseudotime (`pl.dpt_groups_pseudotime`)
 ===========================================================================================================
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.dpt_groups_pseudotime.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.dpt_groups_pseudotime.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2391]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.dpt_groups_pseudotime.html>`__
```

File: tools/scanpy/plot.xml [hunk 11]
```diff
@@ -2364,37 +2364,37 @@ The parameters are the ones of the scatter plot. Call pca_ranking separately
...
 
 Branching trajectories and pseudotime, clustering: Heatmap of pseudotime series (`pl.dpt_timeseries`)
 =====================================================================================================
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.dpt_timeseries.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.dpt_timeseries.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2397]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.dpt_timeseries.html>`__
```

File: tools/scanpy/plot.xml [hunk 12]
```diff
@@ -2407,56 +2407,56 @@ mirrors coordinates along the x axis... that is, you should increase the
 `maxiter` parameter by 1 if the layout is flipped.
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.paga.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.paga.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2410]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.paga.html>`__
```

File: tools/scanpy/plot.xml [hunk 13]
```diff
@@ -2407,56 +2407,56 @@ mirrors coordinates along the x axis... that is, you should increase the
...
 
 
 Branching trajectories and pseudotime, clustering: Scatter and PAGA graph side-by-side (`pl.paga_compare`)
 ==========================================================================================================
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.paga_compare.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.paga_compare.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2417]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.paga_compare.html>`__
```

File: tools/scanpy/plot.xml [hunk 14]
```diff
@@ -2407,56 +2407,56 @@ mirrors coordinates along the x axis... that is, you should increase the
...
 
 Branching trajectories and pseudotime, clustering: Gene expression and annotation changes along paths (`pl.paga_path`)
 ======================================================================================================================
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.paga_path.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.paga_path.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2423]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.paga_path.html>`__
```

File: tools/scanpy/plot.xml [hunk 15]
```diff
@@ -2407,56 +2407,56 @@ mirrors coordinates along the x axis... that is, you should increase the
...
 
 Marker genes: Plot ranking of genes using dotplot plot (`pl.rank_genes_groups`)
 ===============================================================================
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.rank_genes_groups.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.rank_genes_groups.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2429]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.rank_genes_groups.html>`__
```

File: tools/scanpy/plot.xml [hunk 16]
```diff
@@ -2407,56 +2407,56 @@ mirrors coordinates along the x axis... that is, you should increase the
...
 
 Marker genes: Plot ranking of genes as violin plot (`pl.rank_genes_groups_violin`)
 ==================================================================================
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.rank_genes_groups_violin.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.rank_genes_groups_violin.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2435]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.rank_genes_groups_violin.html>`__
```

File: tools/scanpy/plot.xml [hunk 17]
```diff
@@ -2407,56 +2407,56 @@ mirrors coordinates along the x axis... that is, you should increase the
...
 
 Marker genes: Plot ranking of genes as dotplot plot (`pl.rank_genes_groups_dotplot`)
 ====================================================================================
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.rank_genes_groups_dotplot.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.rank_genes_groups_dotplot.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2441]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.rank_genes_groups_dotplot.html>`__
```

File: tools/scanpy/plot.xml [hunk 18]
```diff
@@ -2407,56 +2407,56 @@ mirrors coordinates along the x axis... that is, you should increase the
...
 
 Marker genes: Plot ranking of genes as heatmap plot (`pl.rank_genes_groups_heatmap`)
 ====================================================================================
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.rank_genes_groups_heatmap.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.rank_genes_groups_heatmap.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2447]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.rank_genes_groups_heatmap.html>`__
```

File: tools/scanpy/plot.xml [hunk 19]
```diff
@@ -2407,56 +2407,56 @@ mirrors coordinates along the x axis... that is, you should increase the
...
 
 Marker genes: Plot ranking of genes as matrixplot plot (`pl.rank_genes_groups_matrixplot`)
 ==========================================================================================
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.rank_genes_groups_matrixplot.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.rank_genes_groups_matrixplot.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2453]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.rank_genes_groups_matrixplot.html>`__
```

File: tools/scanpy/plot.xml [hunk 20]
```diff
@@ -2407,56 +2407,56 @@ mirrors coordinates along the x axis... that is, you should increase the
...
 
 Marker genes: Plot ranking of genes as stacked violin plot (`pl.rank_genes_groups_stacked_violin`)
 ==================================================================================================
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pl.rank_genes_groups_stacked_violin.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.rank_genes_groups_stacked_violin.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/plot.xml:2459]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pl.rank_genes_groups_stacked_violin.html>`__
```

File: tools/scanpy/remove_confounders.xml [hunk 21]
```diff
@@ -200,7 +200,7 @@ Correct batch effects with ComBat function (`pp.combat`)
 Corrects for batch effects by fitting linear models, gains statistical power via an EB framework where information is borrowed across genes. This uses the implementation of ComBat
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pp.combat.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pp.combat.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/remove_confounders.xml:203]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pp.combat.html>`__
```

File: tools/scanpy/normalize.xml [hunk 22]
```diff
@@ -341,7 +341,7 @@ The recipe runs the following steps:
 - scale to unit variance and shift to zero mean
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pp.recipe_zheng17.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pp.recipe_zheng17.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/normalize.xml:344]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pp.recipe_zheng17.html>`__
```

File: tools/scanpy/normalize.xml [hunk 23]
```diff
@@ -350,7 +350,7 @@ Normalization and filtering as of Weinreb et al (2017) (`pp.recipe_weinreb17`)
 Expects non-logarithmized data. If using logarithmized data, pass `log=False`.
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pp.recipe_weinreb17.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pp.recipe_weinreb17.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/normalize.xml:353]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pp.recipe_weinreb17.html>`__
```

File: tools/scanpy/normalize.xml [hunk 24]
```diff
@@ -361,7 +361,7 @@ This uses a particular preprocessing.
 Expects non-logarithmized data. If using logarithmized data, pass `log=False`.
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pp.recipe_seurat.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pp.recipe_seurat.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/normalize.xml:364]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pp.recipe_seurat.html>`__
```

File: tools/scanpy/inspect.xml [hunk 25]
```diff
@@ -912,7 +912,7 @@ The returned AnnData object contains:
 This data are stored in the unstructured annotation (uns) and can be accessed using the inspect tool for AnnData objects
 
 More details on the `scanpy documentation
-<https://icb-scanpy.readthedocs-hosted.com/en/stable/api/scanpy.pp.neighbors.html>`__
+<https://scanpy.readthedocs.io/en/stable/api/scanpy.pp.neighbors.html>`__
```
Comment (pavanvidem) [inline tools/scanpy/inspect.xml:915]:
```suggestion
<https://scanpy.readthedocs.io/en/stable/api/generated/scanpy.pp.neighbors.html>`__
```

**Review by pavanvidem — COMMENTED** [review]
I will update the the whole toolsuite soon. Then we will remove them. This time please accept my suggestions. Hopefully, that fixes.

**Comment by pavanvidem** [comment]
Maybe the galaxy version should be bumped now

**Comment by pcm32** [comment]
Apparently those changes did it, thanks @pavanvidem !

**Review by pavanvidem — APPROVED** [review]
I'm not sure whether changing the help is considered a wrapper update or not. Fine for me to approve as it is now.

**Comment by pavanvidem** [comment]
Yes, I mean version suffix. Let's see what others say.

**Comment by bgruening** [comment]
Thanks @pcm32

**Pull request merged**